src/test/suite/participant/participant.test.ts

```diff
@@ -0,0 +1,312 @@
+          prompt: 'hi',
+          response: 'hello',
+        },
+      ],
+    };
+    chatStreamStub = {
+      markdown: sandbox.fake(),
+      button: sandbox.fake(),
+    };
+    chatTokenStub = {
+      onCancellationRequested: () => {},
+    };
+    // The model returned by vscode.lm.selectChatModels is always undefined in tests.
+    sandbox.replace(
+      vscode.lm,
+      'selectChatModels',
+      sandbox.fake.returns([
+        {
+          id: 'modelId',
+          vendor: 'copilot',
+          family: 'gpt-4o',
+          version: 'gpt-4o-date',
+          name: 'GPT 4o (date)',
+          maxInputTokens: 16211,
+          countTokens: () => {},
+          sendRequest: () =>
+            Promise.resolve({
+              text: [
+                '```javascript\n' +
+                  "use('dbOne');\n" +
+                  "db.getCollection('collOne').find({ name: 'example' });\n" +
+                  '```',
+              ],
+            }),
+        },
+      ])
+    );
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+  });
+
+  test('parses a returned by ai text for database and collection name', function () {
+    const text = 'DATABASE_NAME: my  \nCOLLECTION_NAME: cats';
+    const { databaseName, collectionName } =
+      parseForDatabaseAndCollectionName(text);
+    expect(databaseName).to.be.equal('my');
+    expect(collectionName).to.be.equal('cats');
+  });
+
+  test('parses a returned by ai text for code blocks', function () {
+    const text =
+      '```javascript\n' +
+      "use('test');\n" +
+      "db.getCollection('test').find({ name: 'Shika' });\n" +
+      '```';
+    const code = getRunnableContentFromString(text);
+    expect(code).to.be.equal(
+      "\nuse('test');\ndb.getCollection('test').find({ name: 'Shika' });\n"
+    );
+  });
+
+  suite('when connected', function () {
+    beforeEach(function () {
+      sandbox
+        .stub(testParticipantController, '_shouldAskToConnectIfNotConnected')
+        .returns(false);
+    });
+
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    suite('when has not been shown a welcome message yet', function () {
+      beforeEach(function () {
+        sandbox.replace(
+          testParticipantController._storageController,
+          'get',
+          sandbox.fake.returns(false)
+        );
+      });
+
+      afterEach(function () {
+        sandbox.restore();
+      });
+
+      test('prints the message to chat', async function () {
+        const chatRequestMock = {
+          prompt: 'find all docs by a name example',
+          command: 'query',
+          references: [],
+        };
+        await testParticipantController.chatHandler(
+          chatRequestMock,
+          chatContextStub,
+          chatStreamStub,
+          chatTokenStub
+        );
+        const welcomeMessage = chatStreamStub.markdown.firstCall.args[0];
+        expect(welcomeMessage).to.include('Welcome to MongoDB Participant!');
```

Comment:
Should we add an inverse test here in the `when has been shown a welcome message already` suite?

Reply:
There is already. Look for the `when has been shown a welcome message already` suite.